hw/ip/otbn/util/shared/testgen.py

```diff
@@ -0,0 +1,45 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, TextIO
+
+_GPR_NAMES = [f'x{i}' for i in range(32)]
+_WDR_NAMES = [f'w{i}' for i in range(32)]
+
+
+def write_test_data(inputs: Dict[str, bytes], data_file: TextIO) -> None:
+    '''Write the test input data to DMEM.'''
+    data_file.write('.data\n')
+    for name in inputs:
+        data_file.write('\n')
+        if len(inputs[name]) <= 4:
+            # Buffer should be read as a 32-bit (4-byte) value.
+            align = 4
+            length = 4
+        else:
+            # Buffer should be read as a (sequence of) 256-bit (32-byte) values.
+            align = 32
+            length = ((len(inputs[name]) + 31) // 32) * 32
+        data_file.write(f'.balign {align}\n')
+        data_file.write(f'.globl {name}\n')
+        data_file.write(f'{name}:\n')
+        for i in range(0, length, 4):
+            word = int.from_bytes(inputs[name][i:i + 4], byteorder='little')
```

Comment:
Oh, neat! I didn't know that Python allowed you to run off the end of an array like this.